features/cellular/framework/targets/QUECTEL/BG96/QUECTEL_BG96_CellularStack.cpp

```diff
@@ -218,6 +216,39 @@ nsapi_error_t QUECTEL_BG96_CellularStack::create_socket_impl(CellularSocket *soc
 
             handle_open_socket_response(modem_connect_id, err);
         }
+        if (_stack_type == IPV6_STACK || _stack_type == IPV4V6_STACK) {
```

Comment:
Could you add this check only when IP address is feed to AT command and not copy the whole block?
Around lines 192, 211: _at.write_string("127.0.0.1")?

Reply:
Hi @mirelachirica, Thank you for comment. What is meaning 'when IP address is feed to AT command and not copy the whole block?' Could you explain more detail?

Reply:
I mean to add checks for lines 192 and 211. Something like:

if (_stack_type == IPV4_STACK) {
    _at.write_string("127.0.0.1");
} else if (_stack_type == IPV6_STACK || _stack_type == IPV4V6_STACK) {
    _at.write_string("0:0:0:0:0:0:0:1");
}

This way you do not need to copy the whole block of code starting with QIOPEN, lines: 220-251.

Reply:
Oh, that's a good way. Why did not I think about it? I'll check again, thanks.!